op-acceptor/runner/runner.go

```diff
@@ -688,6 +641,108 @@ func (r *runner) parseTestOutput(output []byte, metadata types.ValidatorMetadata
 	return result
 }
 
+// parseTestEvent parses a single line of test output into a TestEvent
+func parseTestEvent(line []byte) (TestEvent, error) {
+	var event TestEvent
+	err := json.Unmarshal(line, &event)
+	return event, err
+}
+
+// isMainTestEvent checks if the event belongs to the main test or package
+func isMainTestEvent(event TestEvent, mainTestName string) bool {
+	return event.Test == "" || event.Test == mainTestName
+}
+
+// processMainTestEvent handles events for the main test
+func processMainTestEvent(event TestEvent, result *types.TestResult, testStart, testEnd *time.Time,
+	errorMsg *strings.Builder, hasSkip *bool) {
+	switch event.Action {
+	case "start":
```

Comment:
nit: we probably should have constants for start/fail/...

Reply:
Added in https://github.com/ethereum-optimism/infra/pull/257